programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1264,6 +1415,129 @@ static void interrupt_handler(int sig) {
 }
 #endif
 
+#if __linux__
+// support interactions with zfw command line utility
+
+static void diverter_update(const char *ip, uint8_t prefix_len, uint16_t lowport, uint16_t highport, const char *protocol, const char *service_id, const char *action) {
+    int rndm;
+    uv_random(NULL, NULL, &rndm, sizeof(rndm), 0, NULL);
+    unsigned short random_port = 1024 + rndm % (65535 - 1023);
```

Comment:
what happens if port is in use?

Reply:
for zfw in tunnel mode it just cant be set to 0.  A setting of 0 indicates that a matching packet should be passed to the OS and not to directly forwarded to the ziti0 interface. As long as it between 1 and 65535 it does not matter if it is in use.  We could make it any static value between 1 and 65535 and remove the uv_random call.  On ziti-router  in tproxy mode it would matter that it was specific to the tproxy port configured for the service.